File: 240328/softeer-6273.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<cmath>

using namespace std;

int main(int argc, char** argv)
{
    int N, M, K;
    cin >> N >> M >> K;
    
    vector<int> weight(N);
    for(int i=0; i<N; i++){
        cin >> weight[i];
    }
    sort(weight.begin(), weight.end());
    
    vector<vector<int>> w_com;
    do{
        w_com.push_back(weight);
    }while(next_permutation(weight.begin(), weight.end()));

    long long answer = 10000000000LL;
    for(int i=0; i<w_com.size(); i++){
        int index = 0;
        long long candidate = 0;
        for(int j=0; j<K; j++){
            long long sum = 0;
            while(sum + w_com[i][index] <= M){
                sum += w_com[i][index];
                index += 1;
                index = index % N;
            }
            candidate += sum;
        }
        answer = min(answer, candidate);
    }
    cout << answer;
    return 0;
}

File: 240328/softeer-6273.py
from itertools import permutations

N, M, K = map(int, input().split())
weight = [int(num) for num in input().split()]

w_com = list(permutations(weight))

answer = 10**10
for w in w_com:
    index = 0
    candidate = 0
    for i in range(K):
        sum = 0
        while sum + w[index] <= M:
            sum += w[index]
            index += 1
            index = index % N
        candidate += sum
    answer = min(answer, candidate)

print(answer)